x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/rerank/TextSimilarityRankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.rerank;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.inference.InferenceServiceResults;
+import org.elasticsearch.inference.InputType;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.feature.RankFeatureDoc;
+import org.elasticsearch.xpack.core.inference.action.InferenceAction;
+import org.elasticsearch.xpack.core.inference.results.RankedDocsResults;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class TextSimilarityRankFeaturePhaseRankCoordinatorContext extends RankFeaturePhaseRankCoordinatorContext {
+
+    protected final Client client;
+    protected final String inferenceId;
+    protected final String inferenceText;
+    protected final float minScore;
```

Comment:
Maybe we could move `minScore` to `RankFeaturePhaseRankCoordinatorContext` ?

Reply:
Not sure about this, this would assume that all reranking implementation operate with scores. I prefer to leave this for a future implementation, if we see some common logic. Let me know if you have a strong opinion about this.